app/controllers/application_controller.rb

```diff
@@ -1,7 +1,17 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
-  def new; end
+  protect_from_forgery with: :exception, prepend: true
+  before_action :authenticate_teacher!
+  before_action :configure_permitted_parameters, if: :devise_controller?
 
-  def create; end
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[first_name last_name email password password_confirmation])
+    devise_parameter_sanitizer.permit(:sign_in, keys: %i[email password])
+    devise_parameter_sanitizer.permit(:account_update,
+                                      keys: %i[first_name last_name email password current_password
+                                               password_confirmation])
+  end
```

Comment:
Given that `Teachers::RegistrationsController` has a `configure_sign_up_params` method, are you sure this code is still needed?

Reply:
I've hot lost in my own comments. I checked and everything seems to work without sanitizers and callbacks in custom registrations controller. What's more, I just discovered, that previously login upon signup worked, and after the aforementioned change it stopped. So I included sign in method after saving the user in #create. Wonder what will blow up now :) Thank You for being so thorough